OpenRA.Mods.Common/Activities/Air/FlyAttack.cs

```diff
@@ -36,12 +36,15 @@ public FlyAttack(Actor self, Target target)
 		public override Activity Tick(Actor self)
 		{
 			if (!target.IsValidFor(self))
+			{
+				self.QueueActivity(new ReturnToBase(self));
```

Comment:
This still doesn't look right, unfortunately  This appends the RTB at the end of the activity queue, which in theory should mean it is redundant vs the `ReturnOnIdle` trait.  If another activity is blocking then neither is going to run.